enhancements/cluster-logging/cluster-logging-kibana-multitenancy.md

```diff
@@ -0,0 +1,188 @@
+Currently users are unable to share their visualizations since Kibana did not have knowledge on how to access the specific user's Kibana index where the visualizations were stored. They would be able to share them now as Kibana specifies the tenant as part of the visualization path when sharing.
+
+### Implementation Details
+
+#### Assumptions
+
+* The Elasticsearch Proxy will configure roles for users such that they have a tenant pattern defined that matches what Kibana would create for an user logging in.
+* Each user would have only two tenants available to them: Global and their user name based tenant
+
+#### Security
+
+No additional security concerns need to be addressed by this plugin.
+
+### Risks and Mitigations
+
+## Design Details
+
+As part of the Kibana6 image building and distribution, we will provide the OpenDistro plugin for Kibana.
+Installing and configuring that plugin will allow us to provide multitenancy from within Kibana.
+We want to avoid using the `private` tenant as it would be a duplicate of the user tenant and may be confusing when trying to share a visualization.
+
+### kibana.conf
+
+As part of configuring multitenancy for Kibana, we would need to add the following to the Kibana config:
+
+```
+elasticsearch.requestHeadersWhitelist: ["securitytenant","Authorization"]
+opendistro_security.multitenancy.enabled: true
+opendistro_security.multitenancy.tenants.enable_global: true
+opendistro_security.multitenancy.tenants.enable_private: false
+opendistro_security.multitenancy.tenants.preferred: ["Global"]
+opendistro_security.multitenancy.enable_filter: false
+```
+
+### roles.yml
+
+In addition to needing to configure Kibana to allow multitenancy, we need to update the Elasticsearch `roles.yml` to specify `tenant_permissions` for each user's role.
+
+```
+example-role:
+  reserved: false
+  hidden: false
+  cluster_permissions:
+  - "read"
+  - "cluster:monitor/nodes/stats"
+  - "cluster:monitor/task/get"
+  index_permissions:
+  - index_patterns:
+    - "applications.*"
+    dls: ""
+    fls: []
+    masked_fields: []
+    allowed_actions:
+    - "read"
+  tenant_permissions:
+  - tenant_patterns:
+    - "kibana_exampleusername"
+    allowed_actions:
+    - "kibana_all_write"
+  static: false
+```
+
+### Test Plan
+
+#### Unit Testing
+
+* Elasticsearch-Proxy unit tests verify that we are able to create the appropriate `tenant_permissions`
+* Elasticsearch-Operator has no actions to take for Kibana -- the config is built into the image itself
+
+#### Integration and E2E tests
+
+* Tests to verify users have index patterns defined for them upon logging in
+* Tests to verify users cannot see other users' tenants
+* Tests to verify users can share visualizations with other users
+* Tests to verify users can view logs without regression
+
+### Graduation Criteria
+
+#### Tech Preview
+
+* Gather feedback from users rather than just developers
+* End user documentation
+* Sufficient test coverage
+
+#### Tech Preview -> GA
```

Comment:
I don't believe this will ever be TP given we intend and have to deliver as part of the move to ES6